Grammar/python.gram

```diff
@@ -638,8 +638,13 @@ type_alias[stmt_ty]:
 # Type parameter declaration
 # --------------------------
 
-type_params[asdl_type_param_seq*]: '[' t=type_param_seq  ']' {
+type_params[asdl_type_param_seq*]: 
+    | '[' t=type_param_seq  ']' {
         CHECK_VERSION(asdl_type_param_seq *, 12, "Type parameter lists are", t) }
+    | '[' token=']' {
+        RAISE_SYNTAX_ERROR_STARTING_FROM(
+            token, 
+            "At least one type parameter definition is expected")}
```

Comment:
I think this is clearer and more succinct:
```suggestion
            "Type parameter list may not be empty")}
```

Reply:
"May not" sounds a bit "weak" to me... how about:

```suggestion
            "Type parameter list cannot be empty")}
```